cacher/iterator.go

```diff
@@ -0,0 +1,187 @@
+// Copyright 2016 The Vulcan Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cacher
+
+import (
+	"net/url"
+
+	"github.com/Sirupsen/logrus"
+	pmodel "github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/storage/local/chunk"
+	"github.com/prometheus/prometheus/storage/metric"
+)
+
+// SeriesIterator fetches from HTTP cacher endpoint to serve datapoints.
+type SeriesIterator struct {
+	chunkIter  chunk.Iterator
+	curr, last *pmodel.SamplePair
+	list       []pmodel.SamplePair
+	idx        int
+	m          metric.Metric
+	ready      chan struct{}
+	resp       *chunksResp
+}
```

Comment:
I think we decided to do compile time checking of 3rd interface satisfaction...i.e `var _ local.SeriesIterator = &SeriesIterator{}`